qiskit/transpiler/passes/optimization/commutation_analysis.py

```diff
@@ -27,9 +27,11 @@ class CommutationAnalysis(AnalysisPass):
     are grouped into a set of gates that commute.
     """
 
-    def __init__(self):
+    def __init__(self, commutation_checker=None):
```

Comment:
Can you make this "private", I'm concerned that this undocumented and it's a public class. Also the argument `CommutationChecker` isn't public afaict.

Reply:
Done, and also made keyword-only in https://github.com/Qiskit/qiskit/commit/351d5f11958e23d0b489b8138de0cb48a0e6a3c8